From 1f5e35001b6b94074d4777662520da088ae279b9 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Fri, 26 Apr 2013 17:10:12 +0300
Subject: [PORT FROM MAIN] Telephony: Fix issue in notifying disconnect completed

BZ: 44786

Currently, on Inactive state notifyDisconnectCompleted is
called with sendAll set to TRUE. This results in
DisconnectCompleted to all the contexts in the apnlist.
With multiple PDP context, there can be more than
1 apn in the apncontextlist.

With this patch, notifyDisconnectCompleted will be
called with sendAll set to false. This way,
NotifyDisconnectCompleted will be sent for only the
reported apnContext.

Change-Id: Ied81ad6423466def4a2543d8415c2053957fa946
Orig-Change-Id: I37d494225959ca0fcf2f5d18249c012ecfad3cd3
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DataConnection.java   |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DataConnection.java b/src/java/com/android/internal/telephony/dataconnection/DataConnection.java
index 9cfed88..763c0f3 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DataConnection.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DataConnection.java
@@ -985,7 +985,7 @@ public final class DataConnection extends StateMachine {
                     log("DcInactiveState: enter notifyDisconnectCompleted +ALL failCause="
                             + mDcFailCause);
                 }
-                notifyDisconnectCompleted(mDisconnectParams, true);
+                notifyDisconnectCompleted(mDisconnectParams, false);
             }
             if (mDisconnectParams == null && mConnectionParams == null && mDcFailCause != null) {
                 if (DBG) {
-- 
1.7.4.1

